packages/core/src/query.ts

```diff
@@ -75,7 +75,15 @@ function arrayOrValue (vv: any): any[] {
 export function resultSort<T extends Doc> (result: T[], sortOptions: SortingQuery<T>): void {
   const sortFunc = (a: any, b: any): number => {
     for (const key in sortOptions) {
-      const result = typeof a[key] === 'string' ? a[key].localeCompare(b[key]) : a[key] - b[key]
+      let aValue = getNestedValue(key, a)
+      if (typeof aValue === 'object') {
+        aValue = JSON.stringify(aValue)
```

Comment:
Why we need to convert to JSON?

Reply:
Simple way for compare objects. Let me know if you have any ideas

Reply:
fast-equals?

Reply:
How can I use type-based soreters with MongoDB...?

Reply:
I think we should not allow sorting by objects.

Reply:
Why we need this part of code in this case?

Reply:
We can extend the `BuildModeKey` from #576 with the `sortKey` property, for example ...